deploy/template.sh

```diff
@@ -69,23 +69,33 @@ findTempateFiles() {
 }
 
 findAndReplaceVariables() {
+  projectNameRegex="${PROJECT_NAME}.+"
+  namespaceRegex="travella-${NAMESPACE}.+"
+  generalRegex="travella\..+"
   for file in ${TEMPLATES[@]}; do
-    local output=${file%.tpl}
-    cp $file $output
-    info "Building $(basename $file) template to $(basename $output)"
-    for variable in ${VARIABLES[@]}; do
-      local value=${!variable}
-      sed -i -e "s/{{ $variable }}/$value/g" $output;
-      sed -i -e "s/{{$variable}}/$value/g" $output;
-    done
-
-    if [[ $? == 0 ]]; then
-        success "Template file $(basename $file) has been successfuly built to $(basename $output)"
-      else
-        error "Failed to build template $(basename $file)"
+    if [[ $file =~ $projectNameRegex ]] \
```

Comment:
Can you explain like 76-79

Reply:
`projectNameRegex` will contain a regex expression that would either be starting from `travella-backend` or `travella-frontend`, depending on what `PROJECT_NAME` variable contains, and then can have any characters.

`namespaceRegex` will contain a regex expression that would be starting from `travella-[NAMESPACE]`, where `NAMESPACE` is the environment we would be building the images for, example: `production` or `staging`,  and then can have any characters.

`generalRegex` will contain a regex expression that would be starting off with `travella.` (notice the dot),  and then can have any characters.

We have three kinds of deployment objects;
1. Deployment objects for:
    - backend (file name starts from `travella-backend`)
    - frontend (file name starts from `travella-frontend`)
2. Deployment objects:
    - staging (file name starts from `travella-staging`)
    - staging (file name starts from `travella-production`)
3. Deployment objects that are shared between all the other deployment objects (file name starts from `travella.`).

So depending on the `project` or `namespace` we would be building the deployment files for, the appropriate templates would get evaluated, also the general templates, shared between all the deployment objects, would get evaluated.

Reply:
Ok Awesome